### PR TITLE
Refs #ManageSEPTypes, scope fix

### PR DESCRIPTION
**Pull request by nks2109**
### Master Redmine ticket
* (Required!)

### Child Redmine ticket(s)
* ()
* ()

### Local build result

```
bundle exec rake parallel:spec[4]
bundle exec cucumber
```

### Latest rebase/merge tag
*

---

### Data ticket(s) Followup
* (redmine links here - optional)

### Related Pull Requests
* (github links here - optional)

---

#### Peer Review
* (For code review)

#### Functional Testing
* (For testing locally)

#### Deployment
* (for release manager and/or build manager)

### CODE REVIEW

File: app/models/setting.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Setting
 
   def self.get_individual_market_monthly_enrollment_due_on
     setting_record = Setting.where(name: "individual_market_monthly_enrollment_due_on").last
-    setting_record = Setting.create(name: "individual_market_monthly_enrollment_due_on", value: '19') if setting_record.blank?
+    setting_record = Setting.create(name: "individual_market_monthly_enrollment_due_on", value: '15') if setting_record.blank?
```
Comment (raghuramg) [inline app/models/setting.rb:12]: move hard coded value to resource registry

File: app/models/family.rb [hunk 2]
```diff
@@ -600,6 +600,23 @@ def latest_ivl_sep
     special_enrollment_periods.individual_market.order_by(:submitted_at.desc).to_a.detect(&:is_active?)
   end
 
+  def options_for_termination_dates(enrollments)
+    latest_sep = latest_shop_sep || latest_fehb_sep
```
Comment (raghuramg) [inline app/models/family.rb:604]: use enrollments to identify market kind (shop vs fehb)

File: app/models/family.rb [hunk 3]
```diff
@@ -600,6 +600,23 @@ def latest_ivl_sep
...
+    return {} unless latest_sep
+
+    enrollments.inject({}) do |date_hash, enrollment|
+      date_hash[enrollment.id.to_s] = latest_sep.termination_dates(enrollment.effective_on)
+      date_hash
+    end
+  end
+
+  def latest_shop_sep_termination_kinds
```
Comment (raghuramg) [inline app/models/family.rb:613]: pass enrollment as this method argument and use it to identify the market kind

File: app/helpers/insured/families_helper.rb [hunk 4]
```diff
@@ -133,24 +134,22 @@ def qle_link_generator_for_an_existing_qle(qle, link_title=nil)
...
-      case kind
-      when 'date_of_event'
-        options << [qle_date.to_s, kind]
-      when 'first_of_this_month'
-        options << [qle_date.beginning_of_month.to_s, kind]
-      when 'fixed_first_of_next_month'
-        options << [(qle_date.end_of_month + 1.day).to_s, kind]
-      end
+    qle.effective_on_kinds.each do |kind|
+      options << [find_effective_on(qle, qle_date, kind).to_s, kind]
```
Comment (raghuramg) [inline app/helpers/insured/families_helper.rb:141]: can be changed to collect and remove options variable

File: app/models/setting.rb (hunk 1 above)
Comment (saimekala07) [reply app/models/setting.rb:12]: Added resource registry setting

File: app/helpers/insured/families_helper.rb (hunk 4 above)
Comment (saimekala07) [reply app/helpers/insured/families_helper.rb:141]: updated method to use collect

**Pull request merged**